### PR TITLE
First pass at Speech API snippets etc

### PR DESCRIPTION
**Pull request by jskeet**
Very little hand-written code, but examples and some docs.

### CODE REVIEW

File: apis/Google.Cloud.Speech.V1Beta1/Google.Cloud.Speech.V1Beta1.Snippets/SpeechClientSnippets.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    public class SpeechClientSnippets
+    {
+        [Fact]
+        public void SyncRecognize()
+        {
+            var audio = LoadResourceAudio("speech.raw");
+            // Snippet: SyncRecognize
+            SpeechClient client = SpeechClient.Create();
+            var config = new RecognitionConfig { Encoding = AudioEncoding.Linear16, SampleRate = 16000 };
+            var response = client.SyncRecognize(config, audio);
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Speech.V1Beta1/Google.Cloud.Speech.V1Beta1.Snippets/SpeechClientSnippets.cs]: The snippet is clearer without using "var", I think.

Comment (jskeet) [inline apis/Google.Cloud.Speech.V1Beta1/Google.Cloud.Speech.V1Beta1.Snippets/SpeechClientSnippets.cs]: Definitely. Sorry, this evolved from my "let's just give it a go" attempt, and I should have fixed that up...

File: apis/Google.Cloud.Speech.V1Beta1/Google.Cloud.Speech.V1Beta1/RecognitionAudioPartial.cs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        /// Asynchronously constructs a <see cref="RecognitionAudio"/> by downloading data from the given URI.
+        /// </summary>
+        /// <param name="uri">The URI to fetch. Must not be null.</param>
+        /// <returns>A task representing the asynchronous operation. The result will be the newly created RecognitionAudio.</returns>
+        public async static Task<RecognitionAudio> FromUriAsync(Uri uri)
+        {
+            GaxPreconditions.CheckNotNull(uri, nameof(uri));
+            using (var client = new HttpClient())
+            {
+                var bytes = await client.GetByteArrayAsync(uri);
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Speech.V1Beta1/Google.Cloud.Speech.V1Beta1/RecognitionAudioPartial.cs]: .ConfigureAsync(false) ?

File: apis/Google.Cloud.Speech.V1Beta1/Google.Cloud.Speech.V1Beta1/RecognitionAudioPartial.cs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        /// Asynchronously constructs a <see cref="RecognitionAudio"/> by downloading data from the given URI.
+        /// </summary>
+        /// <param name="uri">The URI to fetch. Must not be null.</param>
+        /// <returns>A task representing the asynchronous operation. The result will be the newly created RecognitionAudio.</returns>
+        public async static Task<RecognitionAudio> FromUriAsync(string uri)
+        {
+            GaxPreconditions.CheckNotNull(uri, nameof(uri));
+            using (var client = new HttpClient())
+            {
+                var bytes = await client.GetByteArrayAsync(uri);
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Speech.V1Beta1/Google.Cloud.Speech.V1Beta1/RecognitionAudioPartial.cs]: ConfigureAsync ?

Comment (jskeet) [inline apis/Google.Cloud.Speech.V1Beta1/Google.Cloud.Speech.V1Beta1/RecognitionAudioPartial.cs]: Ooh, yes - I should fix Vision as well, which will have the same problem.

File: apis/Google.Cloud.Speech.V1Beta1/Google.Cloud.Speech.V1Beta1/RecognitionAudioPartial.cs (hunk 2 above)
Comment (chrisdunelm) [inline apis/Google.Cloud.Speech.V1Beta1/Google.Cloud.Speech.V1Beta1/RecognitionAudioPartial.cs]: Or even ConfigureAwait(false)

Comment (jskeet) [inline apis/Google.Cloud.Speech.V1Beta1/Google.Cloud.Speech.V1Beta1/RecognitionAudioPartial.cs]: For simplicity, I'll fix Vision in the same PR (but a separate commit).

**Comment by jskeet** [comment]
(I suspect we may want to look at the use of Task.Yield at some point. It doesn't do what I'd wanted, which I should have realized.)

**Comment by chrisdunelm** [comment]
LGTM

**Pull request merged**